hack/run-functional-tests.sh

```diff
@@ -73,107 +73,171 @@ check_arguments() {
+
+                    echo "[ERROR] Test did not pass. Exiting..."
+                    exit 1
 
-                  if [ "${test_type}"  == "pass" ] && [ ${exit_code} -ne 0 ]; then
+                elif [ "${test_type}"  == "fail" ] && [ ${exit_code} -ne 1 ]; then
 
-                      echo "[ERROR] Test did not pass. Exiting..."
-                      exit 1
+                    echo "[ERROR] Test did not pass. Exiting..."
+                    exit 1
 
-                  elif [ "${test_type}"  == "fail" ] && [ ${exit_code} -ne 1 ]; then
+                else
 
-                      echo "[ERROR] Test did not pass. Exiting..."
-                      exit 1
+                    echo "[INFO] Test Passed"
 
-                  else
+                fi
 
-                      echo "[INFO] Test Passed"
+            fi
 
-                  fi
+        else
 
-              fi
+            echo "[WARN] File \"${test_file_path}\" not found. Skipping..."
 
-          else
+        fi
 
-              echo "[WARN] File \"${test_file_path}\" not found. Skipping..."
+        # check to see if the user specified a script to be associated with this stanza
+        # only run script for apply actions
+        if [[ ${user_script} != "" ]] && [[ "${action}" == "apply" ]]; then
 
-          fi
+          # if they did specify a script run it with the between argument and pass the namespace in
+          ${user_script_path} "between" "${TEST_NAMESPACE}"
 
-          echo "============================================================================"
-          
-      done
+        fi
+
+        echo "============================================================================"
+        
+    done
+
+    # check to see if the user specified a script to be associated with this stanza
+    # only run script for apply actions
+    if [[ ${user_script} != "" ]] && [[ "${action}" == "apply" ]]; then
+
+      # if they did specify a script run it with the teardown argument and pass the namespace in
+      ${user_script_path} "teardown" "${TEST_NAMESPACE}"
+
+      echo "============================================================================"
 
     fi
 
+  fi
+
 }
 
 # **********************************************
 # Determine test scope
 # **********************************************
 scope_and_run_tests() {
 
+  #create identifiable local variable for argument 1, the action to perform
   local action="${1}"
 
-  if [ "${TEST_RESOURCE_TYPE}" == "all" ]; then
-
-    resources=$(yq read "${TESTS_MANIFEST}" 'resources.[*].name')
+  #size the array of resources
+  local resource_array_length
+  resource_array_length=$(yq read -l "${TESTS_MANIFEST}" 'resources')
 
-    for resource in ${resources}; do
 
-      if [ "${TEST_TYPE}" == "all" ]; then
+  #loop through all resources in the supplied manifest
+    #determine which indicies meet the supplied criteria in ${TEST_RESOURCE_KIND} and ${TEST_RESOURCE_DESIRED}
```

Comment:
Consistency nit, space after # for comments.

Reply:
Also, try and align comments to the code block they're referencing. Not sure if it's my monitor, the way Github is displaying it, or what, but it looks misaligned here.